File: java-engine/src/main/java/hu/elteik/knowledgelab/javaengine/algorithms/rotor_router_with_leader_dispersion/models/RotorRouterWithLeaderDispersionNode.java
package hu.elteik.knowledgelab.javaengine.algorithms.rotor_router_with_leader_dispersion.models;

import hu.elteik.knowledgelab.javaengine.core.models.Node;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper=false)
@AllArgsConstructor
@NoArgsConstructor
public class RotorRouterWithLeaderDispersionNode extends Node {
    
    private Long rotorRouter;

}

File: dispersion-app/src/app/models/algorithms/rotor-router-with-leader-dispersion/RotorRouterWithLeaderDispersionNode.ts
import { Node } from '../../core/Node';
import { NodeState } from '../../utils/NodeState';

export class RotorRouterWithLeaderDispersionNode extends Node {
  rotorRouter: number;

  constructor(id: number, state: NodeState, rotorRouter: number) {
    super(id, state);
    this.rotorRouter = rotorRouter;
  }
}
